spiketools/spatial/place.py

```diff
@@ -0,0 +1,128 @@
+"""Place cell related functions."""
+
+import numpy as np
+
+from spiketools.spatial.utils import compute_nbins
+from spiketools.spatial.occupancy import compute_occupancy, compute_bin_counts_pos
+from spiketools.utils.extract import (get_range, get_values_by_time_range, get_values_by_times,
+                                      threshold_spikes_by_values)
+
+###################################################################################################
+###################################################################################################
+
+def compute_place_bins(spikes, position, timestamps, bins, area_range=None,
+                       speed=None, speed_threshold=None, time_threshold=np.inf,
+                       occupancy=None):
+    """Compute the spatially binned spiking activity.
+
+    Parameters
+    ----------
+    spikes : 1d array
+        Spike times, in seconds.
+    position : 1d or 2d array
+        Position values.
+    timestamps : 1d array
+        Timestamps.
+    bins : int or list of [int, int]
+        The bin definition for dividing up the space. If 1d, can be integer.
+        If 2d should be a list, defined as [number of x_bins, number of y_bins].
+    area_range : list of list, optional
+        Edges of the area to bin, defined as [[x_min, x_max], [y_min, y_max]].
+    speed : 1d array
+        Current speed for each position.
+        Should be the same length as timestamps.
+    speed_threshold : float, optional
+        Speed threshold to apply.
+        If provided, any position values with an associated speed below this value are dropped.
+    time_threshold : float, optional
+        A maximum time threshold, per bin observation, to apply.
+        If provided, any bin values with an associated time length above this value are dropped.
+    occupancy : 1d or 2d array
```

Comment:
`occupancy` should be optional if it's optional in `compute_bin_counts_pos` function, right?